### PR TITLE
[KOGITO-5313][KOGITO-5315] Upgrade Kogito operator and BDD tests to Infinispan 12

### PR DESCRIPTION
**Pull request by sutaakar**
Many thanks for submitting your Pull Request :heart:! 

https://issues.redhat.com/browse/KOGITO-5313
https://issues.redhat.com/browse/KOGITO-5315

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

**Review by radtriste — APPROVED** [review]

File: go.sum [hunk 1]
```diff
@@ -154,6 +154,8 @@ github.com/DataDog/zstd v1.3.6-0.20190409195224-796139022798/go.mod h1:1jcaCB/uf
 github.com/DataDog/zstd v1.4.1/go.mod h1:1jcaCB/ufaK+sKp1NBhlGmpz41jOoPQ35bpF36t7BBo=
 github.com/Djarvur/go-err113 v0.0.0-20200410182137-af658d038157/go.mod h1:4UJr5HIiMZrwgkSPdsjy2uOQExX/WEILpIrO9UPGuXs=
 github.com/Djarvur/go-err113 v0.1.0/go.mod h1:4UJr5HIiMZrwgkSPdsjy2uOQExX/WEILpIrO9UPGuXs=
+github.com/GeertJohan/go.incremental v1.0.0/go.mod h1:6fAjUhbVuX1KcMD3c8TEgVUqmo4seqhv0i0kdATSkM0=
```
Comment (spolti) [inline go.sum:157]: for what these two dependencies are used?

**Review by spolti — APPROVED** [review]
approving, left a comment about deps added.

File: go.sum (hunk 1 above)
Comment (sutaakar) [reply go.sum:157]: I guess it is something internal brought by Infinispan dependency upgrade. It was automatically added when I built the operator.

**Review by ricardozanini — APPROVED** [review]
Well done!

**Pull request merged**